Janky worked correctly:
```
19:16:37 ---> Making bundle: test-integration-flaky (in bundles/test-integration-flaky)
19:16:38 
19:16:38 Found new integrations tests:
19:16:38 +func TestCommitFlakyRandom1(t *testing.T) {
19:16:38 +func TestCommitFlakyRandom2(t *testing.T) {
19:16:38 Running stress test for them.
19:16:38 Using test flags: -test.count 5 -test.run TestCommitFlakyRandom1|TestCommitFlakyRandom2
.
.
.
19:18:08 Running /go/src/github.com/docker/docker/integration/image
19:18:08 INFO: Testing against a local daemon
19:18:08 === RUN   TestCommitFlakyRandom1
19:18:08 --- PASS: TestCommitFlakyRandom1 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom2
19:18:08 --- PASS: TestCommitFlakyRandom2 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom1
19:18:08 --- PASS: TestCommitFlakyRandom1 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom2
19:18:08 --- PASS: TestCommitFlakyRandom2 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom1
19:18:08 --- FAIL: TestCommitFlakyRandom1 (0.00s)
19:18:08     commit_test.go:61: assertion failed: error is not nil: Game Over, Better Luck Next Time
19:18:08 === RUN   TestCommitFlakyRandom2
19:18:08 --- PASS: TestCommitFlakyRandom2 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom1
19:18:08 --- PASS: TestCommitFlakyRandom1 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom2
19:18:08 --- PASS: TestCommitFlakyRandom2 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom1
19:18:08 --- PASS: TestCommitFlakyRandom1 (0.00s)
19:18:08 === RUN   TestCommitFlakyRandom2
19:18:08 --- PASS: TestCommitFlakyRandom2 (0.00s)
19:18:08 FAIL
```

Others needs some tweaking.

--- a/Makefile
+++ b/Makefile
@@ -165,6 +165,9 @@ test-integration-cli: test-integration ## (DEPRECATED) use test-integration
 test-integration: build ## run the integration tests
 	$(DOCKER_RUN_DOCKER) hack/make.sh dynbinary test-integration
 
+test-integration-flaky: build ## run the stress test for all new integration tests
+	$(DOCKER_RUN_DOCKER) hack/make.sh dynbinary test-integration-flaky
+
 test-unit: build ## run the unit tests
 	$(DOCKER_RUN_DOCKER) hack/test/unit
 

--- a/hack/ci/janky
+++ b/hack/ci/janky
@@ -13,5 +13,6 @@ hack/make.sh \
 	binary-daemon \
 	dynbinary \
 	test-docker-py \
+	test-integration-flaky \
 	test-integration \
 	cross

--- a/hack/make/test-integration-flaky
+++ b/hack/make/test-integration-flaky
@@ -1,0 +1,26 @@
+#!/usr/bin/env bash
+set -e -o pipefail
+
+source hack/validate/.validate
+new_tests=$(
+    validate_diff --diff-filter=ACMR --unified=0 -- 'integration/*_test.go' |
+    grep -E '^(\+func )(.*)(\*testing)' || true
+)
+
+if [ -z "$new_tests" ]; then
+    echo 'No new tests added to integration.'
+    return
+fi
+
+echo
+echo "Found new integrations tests:"
+echo "$new_tests"
+echo "Running stress test for them."
+
+(
+    TESTARRAY=$(echo "$new_tests" | sed 's/+func //' | awk -F'\\(' '{print $1}' | tr '\n' '|')
+    export TESTFLAGS="-test.count 5 -test.run ${TESTARRAY%?}"
+    export TEST_REPEAT=5
+    echo "Using test flags: $TESTFLAGS"
+    source hack/make/test-integration
+)

